docker/elg_stance.py

```diff
@@ -131,8 +140,9 @@ async def process_request():
         classification, scores = await run_sync(classifier.classify)(original.json, embedded.json)
 
         features = stance_classification_as_features(classification, scores)
-        features["tweet_id"] = tweet_id
-        features["in_reply_to"] = in_reply_to
+        if add_id_features:
```

Comment:
This is the only bit I'm unsure about - I'm deliberately _not_ adding the tweet ID and in-reply-to-ID features in structuredText mode because (a) sometimes they will be generated ones rather than real ones, which don't mean anything to the caller, and (b) the start/end offsets tell you which text the result relates to and you can easily cross-reference that with the request

Reply:
or is it better to leave them in if the request had explicit IDs included (rather than relying on the pattern of original first, replies after)?

Reply:
Hmm, if the IDs are explicitly included in the request I'd be tempted to put them in the output otherwise callers may need to then look back at the input they sent to work out the ID for a given TweetStance annotation. I'm envisaging a situation where the tweets have come from a DB and we want to update them with the stance feature. If the ID is in the annotation then updating the record would be easy, if not you'd have to look back at the input to work out what it was.

In fact given we are changing things I'm wondering if we should also remove references to tweet from the output (so annotaiton would be `Stance` and features `id` and `in_reply_to`.  Although if we were going that far then I might also want to add support for `id` and `in_reply_to` as the fields in the input as well as the twitter specific ones (i.e. if `id_str` isn't present then try `id` instead etc.).

Reply:
> if the IDs are explicitly included in the request I'd be tempted to put them in the output otherwise callers may need to then look back at the input they sent to work out the ID for a given TweetStance annotation.

Yeah, I think you're right there - I've changed it so that we will add the `tweet_id` and `in_reply_to` if they were there explicitly in the input (heuristically - I'm suppressing any ID that starts `gen:` as that's the format I've used for the auto-generated IDs).

> In fact given we are changing things I'm wondering if we should also remove references to tweet from the output (so annotaiton would be `Stance` and features `id` and `in_reply_to` regardless of which version of code runs)

Changing the annotation type would break any existing users of the cloud service (unless we mapped Stance back to TweetStance at the xgapp level).  Support for other input features seems to me to go beyond the scope of this PR so if you're happy I'll merge this now and then think about that separately.

Reply:
Yes, let's leave at this for now, and we can think about other changes later.